### PR TITLE
Changed title_text and issue #4547

### PR DESCRIPTION
**Pull request by HuntJSparra**
Replaced some deprecated title_parameter with the new title.parameter.

Solves issues #4547 also

### CODE REVIEW

File: bokeh/core/compat/bokeh_renderer.py [hunk 1]
```diff
@@ -82,7 +83,7 @@ def close_figure(self, fig):
                              plot_width=self.width,
                              plot_height=self.height)
 
-                _plot.title = ""
+                _plot.title = Title()
```
Comment (bryevdv) [inline bokeh/core/compat/bokeh_renderer.py]: Plots have a default Title object. You can just set _plot.title.text no need to create a new Title

**Comment by HuntJSparra** [comment]
ping @bryevdv

File: bokeh/core/compat/bokeh_renderer.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
                      Grid, GridPlot, LinearAxis, Plot, CategoricalAxis, Legend)
 from ...models.glyphs import (Asterisk, Circle, Cross, Diamond, InvertedTriangle,
                             Line, MultiLine, Patches, Square, Text, Triangle, X)
+from ...models.annotations import Title
```
Comment (bryevdv) [inline bokeh/core/compat/bokeh_renderer.py]: I think this import is unneeded / unused now

**Comment by bryevdv** [comment]
One minor comment otherwise LGTM

**Comment by bryevdv** [comment]
Hey @HuntJSparra just for expediency I am going to merge this to close #4547, I'll clean up the unused import in another PR.

**Pull request merged**